### PR TITLE
Specify a confidence threshold (green|amber|red) when loading PanelApp panels

### LINKED ISSUE #3536: PanelApp panels: both high confidence (green) and lower confidence (green, amber and red)
Hi all!

One of our collaborators using scout asked after a possible feature request.

Basically implementing the ability to have a two tier system to first filter for variants by a high confidence PanelApp panel (comprised of genes on the green list). Then if nothing is found, having the ability to filter for variants by a PanelApp panel which includes lower confidence genes (comprised of genes on the green, amber and red lists). As far as I understand, only genes on the high confidence/green list for PanelApp panels are currently utilised in scout.

Apologies if this feature is already available in scout somewhere, I couldn't find anything along these lines in the docs or admin configuration files.

What are your thoughts?

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #3536  --> Allow loading PanelApp panels with different confidence thresholds. Green if confidence is not specified:

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>

**How to test, locally**:
2. Try loading one more PanelApp: `scout --demo load panel --panel-app --panel-id 477 (--panel-app-confidence amber)`
3. Try loading all panels: `scout --demo load panel --panel-app  (--panel-app-confidence red)`

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DNIL
- [x] tests executed by CR, LK

### CODE REVIEW

File: scout/commands/load/panel.py [hunk 1]
```diff
@@ -101,53 +105,3 @@ def panel(
     except Exception as err:
         LOG.warning(err)
         raise click.Abort()
-
-    return
-
-
-def _panel_omim(adapter, genemap2, mim2genes, api_key, institute):
```
Comment (northwestwitch) [inline scout/commands/load/panel.py:108]: I've just moved this to a better place, together with similar panel loading helpers

**Comment by northwestwitch** [comment]
I don't know how to fix the code smell found by SonarCloud in a way that is not what I wrote in #3564

File: scout/load/panel.py [hunk 2]
```diff
@@ -143,3 +149,47 @@ def load_panel_app(adapter, panel_id=None, institute="cust000"):
             adapter.load_panel(parsed_panel=parsed_panel, replace=True)
         except Exception as err:
             raise err
+
+
+def load_omim_panel(adapter, genemap2, mim2genes, api_key, institute):
+    """Add OMIM panel to the database.
```
Comment (northwestwitch) [inline scout/load/panel.py:155]: Moved here basically unchanged

**Comment by northwestwitch** [comment]
Hi @leahkemp, I made a PR with the feature you requested some time ago. You could test the branch and see how you like it if you want?

**Comment by leahkemp** [comment]
Woohoo! I'll give it a try today and try get back to you quickly for merging! :)

**Comment by leahkemp** [comment]
Thank's all! It's looking great to me! Here's a brain dump of my testing

## Setup

```bash
# install scout in conda env
git clone https://github.com/Clinical-Genomics/scout
cd scout
git checkout panelapp_other_confidence
mamba create --name test_scout_env python=3.10.4
conda activate test_scout_env
pip install --requirement requirements.txt --editable .
scout -db test-database setup database --api-key big_secret --institute-name test
```

## Test on demo scout instance

### Try loading one panel

```bash
scout --demo load panel --panel-app --panel-id 477 --panel-app-confidence amber
scout --demo load panel --panel-app --panel-id 477 --panel-app-confidence green
scout --demo load panel --panel-app --panel-id 477 --panel-app-confidence red
```

Code ran fine and I can see the panel in the scout GUI. But not the expected number of genes? I guess it's grabbing from a reduced test/demo dataset?

panels_loaded_in_scout

### Try loading all panels

```bash
scout --demo load panel --panel-app --panel-app-confidence red
scout --demo load panel --panel-app --panel-app-confidence amber
scout --demo load panel --panel-app --panel-app-confidence green
```

Code ran fine and I can see the panels in the scout GUI

panels_loaded_in_scout_2

## Test on full scout instance

### Try loading one panel

```bash
scout -db test-database load panel --panel-app --panel-id 477 --panel-app-confidence amber --institute test
scout -db test-database load panel --panel-app --panel-id 477 --panel-app-confidence green --institute test
scout -db test-database load panel --panel-app --panel-id 477 --panel-app-confidence red --institute test
```

### Try loading all panels

```bash
scout -db test-database load panel --panel-app --panel-app-confidence amber --institute test
scout -db test-database load panel --panel-app --panel-app-confidence green --institute test
scout -db test-database load panel --panel-app --panel-app-confidence red --institute test
```

Code ran fine and I can see the panels in the scout GUI

panels_loaded_in_scout_3

Had a little play filtering the variants in the demo scout instance by the PanelApp panels, looks to be working fine :)

**Comment by northwestwitch** [comment]
> ```shell
> ```shell
> scout --demo load panel --panel-app --panel-id 477 --panel-app-confidence amber
> scout --demo load panel --panel-app --panel-id 477 --panel-app-confidence green
> scout --demo load panel --panel-app --panel-id 477 --panel-app-confidence red
> ```
> 
> 
>     
>       
>     
> 
>       
>     
> 
>     
>   
> Code ran fine and I can see the panel in the scout GUI. But not the expected number of genes? I guess it's grabbing from a reduced test/demo dataset?
> ```

Thanks for the testing and the nice report! It shouldn't be collecting genes from a demo file, so no, it should be all genes?! 🤔 

I'll try to figure out what's wrong here!

**Comment by northwestwitch** [comment]
> ```shell
> --panel-app --panel-id 477 --panel-app-confidence green
> ```

Ah I get it now, most genes are missing in the demo instance, so when you populate the panel they are skipped, and you get this:

image


All good then!

File: docs/admin-guide/panelapp_panels.md [hunk 3]
```diff
@@ -4,15 +4,32 @@ Gene panels can be imported from [PanelApp](https://panelapp.genomicsengland.co.
...
+
+Please note that when you create a PanalApp panel in **Scout, the software uses the specified level as a threshold, and not as a filter**, so you'll get the following:
+
+```
+`--panel-app-confidence green` collects only green genes from PanelApp
+`--panel-app-confidence amber` collects amber and green genes from PanelApp
+`--panel-app-confidence red` collects all genes from PanelApp
+```
+
+Please note that **if you don't specify any confidence level with the `--panel-app-confidence option`, then only the `HighEvidence` (green) genes will be included in the panels**.
```
Comment (dnil) [inline docs/admin-guide/panelapp_panels.md:24]: 💯

File: docs/admin-guide/panelapp_panels.md [hunk 4]
```diff
@@ -4,15 +4,32 @@ Gene panels can be imported from [PanelApp](https://panelapp.genomicsengland.co.
 
 The command to **import/update all gene panels from PanelApp** is the following:
 
-`scout load panel --panel-app  --institute <inst-id>`
+`scout load panel --panel-app  --institute <inst-id> (--panel-app-confidence green|amber|red)`
```
Comment (dnil) [inline docs/admin-guide/panelapp_panels.md:7]:
```suggestion
`scout load panel --panel-app  --institute <inst-id> [--panel-app-confidence <green|amber|red>]`
```

File: docs/admin-guide/panelapp_panels.md [hunk 5]
```diff
@@ -4,15 +4,32 @@ Gene panels can be imported from [PanelApp](https://panelapp.genomicsengland.co.
...
+`--panel-app-confidence amber` collects amber and green genes from PanelApp
+`--panel-app-confidence red` collects all genes from PanelApp
+```
+
+Please note that **if you don't specify any confidence level with the `--panel-app-confidence option`, then only the `HighEvidence` (green) genes will be included in the panels**.
 
 To **create/update only one** PanelApp panel, use the following command:
 
-`scout load panel --panel-app  --institute <inst-id> --panel-id <panel-id>`.
+`scout load panel --panel-app  --institute <inst-id> --panel-id <panel-id> (--panel-app-confidence green|amber|red)`.
```
Comment (dnil) [inline docs/admin-guide/panelapp_panels.md:28]:
```suggestion
`scout load panel --panel-app  --institute <inst-id> --panel-id <panel-id> [--panel-app-confidence <green|amber|red>]`.
```

File: scout/parse/panel.py [hunk 6]
```diff
@@ -328,25 +333,27 @@ def parse_panel_app_panel(panel_info, hgnc_map, institute="cust000", panel_type=
...
+    )
     gene_panel["institute"] = institute
     gene_panel["panel_type"] = panel_type
 
     LOG.info("Parsing panel %s", gene_panel["display_name"])
 
     gene_panel["genes"] = []
 
-    nr_low_confidence = 1
+    nr_excluded = 0
```
Comment (dnil) [inline scout/parse/panel.py:346]: 👍

File: scout/parse/panel.py [hunk 7]
```diff
@@ -328,25 +333,27 @@ def parse_panel_app_panel(panel_info, hgnc_map, institute="cust000", panel_type=
...
+        [panel_info["SpecificDiseaseName"], f"[{confidence.upper()}]"]
+    )
     gene_panel["institute"] = institute
     gene_panel["panel_type"] = panel_type
 
     LOG.info("Parsing panel %s", gene_panel["display_name"])
 
     gene_panel["genes"] = []
 
-    nr_low_confidence = 1
```
Comment (dnil) [inline scout/parse/panel.py:339]: 👀

**Review by dnil — APPROVED** [review]
Function spec looks good, code looks good; couple of minor suggestions for the docs format. I skipped testing since both you and Leah already did!

**Comment by northwestwitch** [comment]
Merging as soon as the tests pass, thanks both!

**Pull request merged**